### PR TITLE
GAP-35 CUDA compute-capability property

### PR DESCRIPTION
**Pull request by pwalski**
Resolves: https://github.com/golemfactory/gamerhash-facade/issues/56

### CODE REVIEW

File: gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md [hunk 1]
```diff
@@ -62,11 +62,12 @@ Namespace that describes GPU CUDA capabilities
...
+| `golem.!exp.gap-35.v1.inf.gpu.cuda.version`             | `string`  | Offer      | A version of CUDA that is guaranteed to work. The range of compatibile versions is not known by the GPU because there is a minimum CUDA version and also it can be deprecated in a future CUDA version. It can also be different for CUDA and a given library, for example tensorflow had a different GPU compatibility requirement for building from source and using prebuild binaries. Get the info from Nvidia and/or a general guideline from StackOverflow. The property is a version string including a major version, and a minor version in #.# format. |
+| `golem.!exp.gap-35.v1.inf.gpu.cuda.compute-capability`  | `string`  | Offer      | Every [Nvidia GPU](https://developer.nvidia.com/cuda-gpus) has Compute Capability level. It allows to identify [CUDA features it supports](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities). The property is a version string including a major version, and a minor version in #.# format. |
```
Comment (pwalski) [inline gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md:70]: New `compute-capability` property.

File: gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md [hunk 2]
```diff
@@ -62,11 +62,12 @@ Namespace that describes GPU CUDA capabilities
...
+| ------------------------------------------------------- | --------- | ---------- | ----------- |
+| `golem.!exp.gap-35.v1.inf.gpu.cuda.enabled`             | `boolean` | Offer      | this is related to the image running and not the VM runtime |
+| `golem.!exp.gap-35.v1.inf.gpu.cuda.cores`               | `integer` | Offer      | This can be found from the general specification of each GPU. Otherwise it is platform specific that we would not like to work with. We should aim to focus on platform independent metrics that are easy to access in a health check |
+| `golem.!exp.gap-35.v1.inf.gpu.cuda.version`             | `string`  | Offer      | A version of CUDA that is guaranteed to work. The range of compatibile versions is not known by the GPU because there is a minimum CUDA version and also it can be deprecated in a future CUDA version. It can also be different for CUDA and a given library, for example tensorflow had a different GPU compatibility requirement for building from source and using prebuild binaries. Get the info from Nvidia and/or a general guideline from StackOverflow. The property is a version string including a major version, and a minor version in #.# format. |
```
Comment (pwalski) [inline gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md:69]: Added last sentence with info about format.

File: gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md [hunk 3]
```diff
@@ -76,21 +77,21 @@ Namespace that describes information about GPU clocks
...
-| `golem.!exp.gap-35.v1.inf.gpu.clock.sm.mhz`       | `integer` | Offer      | The max rate of the streaming multiprocessor clock as reported by nvidia-smi. CUDA cores are driven by this clock. |
-| `golem.!exp.gap-35.v1.inf.gpu.clock.video.mhz`    | `integer` | Offer      | The max rate of the video clock as reported by nvidia-smi                                                          |
+| `golem.!exp.gap-35.v1.inf.gpu.clocks.graphics.mhz` | `integer` | Offer      | The max rate of the graphics clock as reported by nvidia-smi                                                       |
+| `golem.!exp.gap-35.v1.inf.gpu.clocks.memory.mhz`   | `integer` | Offer      | The max rate of the memory clock as reported by nvidia-smi                                                         |
+| `golem.!exp.gap-35.v1.inf.gpu.clocks.sm.mhz`       | `integer` | Offer      | The max rate of the streaming multiprocessor clock as reported by nvidia-smi. CUDA cores are driven by this clock. |
+| `golem.!exp.gap-35.v1.inf.gpu.clocks.video.mhz`    | `integer` | Offer      | The max rate of the video clock as reported by nvidia-smi                                                          |
```
Comment (pwalski) [inline gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md:83]: Unification of property. Implementation and a part of this GAP had `clocks` so I updated the name only here.

File: gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md [hunk 4]
```diff
@@ -76,21 +77,21 @@ Namespace that describes information about GPU clocks
...
 
 #### Properties
 
-| Property                                             | Type      | Applies to | Description                                                               |
-| ---------------------------------------------------- | --------- | ---------- | ------------------------------------------------------------------------- |
-| `golem.!exp.gap-35.v1.inf.gpu.memory.bandwidth.gib` | `integer` | Offer      | the theoretical maximum amount of data that the bus can handle per second |
-| `golem.!exp.gap-35.v1.inf.gpu.memory.total.gib`     | `integer` | Offer      | indicates the amount of memory available to the GPU                       |
+| Property                                            | Type      | Applies to | Description                                                               |
+| --------------------------------------------------- | --------- | ---------- | ------------------------------------------------------------------------- |
+| `golem.!exp.gap-35.v1.inf.gpu.memory.bandwidth.gib` | `integer` | Offer      | **Optional** The theoretical maximum amount of data that the bus can handle per second |
```
Comment (pwalski) [inline gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md:93]: Marked as optional

File: gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md (hunk 3 above)
Comment (pwalski) [reply gaps/gap-35_gpu_pci_capability/gap-35_gpu_pci_capability.md:83]: `ya-runtime-vm-nvidia` has `clocks`, so we're good:
https://github.com/golemfactory/ya-runtime-vm-nvidia/blob/257ee60195996bc69f16d3721e5dbea371ab8495/self_test_img/src/lib.rs#L181

**Review by nieznanysprawiciel — APPROVED** [review]

**Pull request merged**